gui/widgets/editable.cpp

```diff
@@ -389,4 +503,19 @@ void EditableWidget::makeCaretVisible() {
 	drawCaret(false);
 }
 
+void EditableWidget::initHighlight() {
+	// disable highlight
+	_highlightVisible = false;
+	// Empty the temporary string of previous values
+	_highlightString.clear();
+	// insert blank characters
+	for (int i = 0; i < (int)_caretPos; i++)
+		_highlightString.insertChar(' ', i);
```

Comment:
What is the logic behind this?

Reply:
I have tried using append() method to add characters, but had difficulty in reversing the string and displaying the same. So the logic thought was to have an empty temporary string equal to the size of the caret position and add characters from the end of the temporary string.